File: trees/sum_of_left_leaves.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        if root is None:
            return 0

        sum_left_leaves = 0

        if root.left and root.left.left is None and root.left.right is None:
            sum_left_leaves += root.left.val

        sum_left_leaves += self.sumOfLeftLeaves(root.left) 
        sum_left_leaves += self.sumOfLeftLeaves(root.right)

        return sum_left_leaves

File: trees/sum_of_left_leaves.rb
# Definition for a binary tree node.
# class TreeNode
#     attr_accessor :val, :left, :right
#     def initialize(val = 0, left = nil, right = nil)
#         @val = val
#         @left = left
#         @right = right
#     end
# end
# @param {TreeNode} root
# @return {Integer}
def sum_of_left_leaves(root)
    return 0 if root.nil?

    sum_left_leaves = 0

    if root.left && root.left.left.nil? && root.left.right.nil?
        sum_left_leaves += root.left.val
    end

    sum_left_leaves += sum_of_left_leaves(root.left)
    sum_left_leaves += sum_of_left_leaves(root.right)

    sum_left_leaves
end